runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/SplittableParDo.java

```diff
@@ -391,10 +392,10 @@ public String translateRestrictionCoderId(SdkComponents newComponents) {
    * collection is effectively the same elements as input, but the per-key state and timers are now
    * effectively per-element.
    */
-  private static class RandomUniqueKeyFn<T> implements SerializableFunction<T, String> {
+  private static class RandomUniqueKeyFn<T> implements SerializableFunction<T, byte[]> {
     @Override
-    public String apply(T input) {
-      return UUID.randomUUID().toString();
+    public byte[] apply(T input) {
+      return UUID.randomUUID().toString().getBytes(StandardCharsets.UTF_8);
```

Comment:
nit: you can convert to bytes directly via https://stackoverflow.com/questions/17893609/convert-uuid-to-byte-that-works-when-using-uuid-nameuuidfrombytesb